### PR TITLE
[v1.7] Cherry-pick #17776 #17681

### LINKED ISSUE #18462: [v1.7] PyPI description of nightly wheels outdated
The PyPI description for v1.7 nightly wheels is outdated. The wheels use the non MKL-DNN versions of doc files for each flavor.

This is where doc files are referenced:
https://github.com/apache/incubator-mxnet/blob/v1.7.x/tools/pip/setup.py#L116

The doc folder needs to be updated as per #17776 #17681 
https://github.com/apache/incubator-mxnet/tree/v1.7.x/tools/pip/doc

### PR DESCRIPTION
**Pull request by mseth10**
## Description ##
Cherry-pick #17776 #17681 to update PyPI description

### CODE REVIEW

**Comment by mseth10** [comment]
@mxnet-bot run ci [unix-gpu, miscellaneous]

**Comment by mseth10** [comment]
@mxnet-bot run ci [unix-gpu]

File: cd/mxnet_lib/dynamic/Jenkins_pipeline.groovy [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 // NOTE: the following variables are referenced in the mxnet_lib_pipeline jenkins file imported bellow
 // libmxnet location
-libmxnet = 'lib/libmxnet.so'
+libmxnet = 'build/libmxnet.so'
```
Comment (leezu) [inline cd/mxnet_lib/dynamic/Jenkins_pipeline.groovy:27]: This is for switching to the cmake build on CD. Should it be done separately / only for 1.x?

Comment (leezu) [reply cd/mxnet_lib/dynamic/Jenkins_pipeline.groovy:27]: @mseth10

Comment (mseth10) [reply cd/mxnet_lib/dynamic/Jenkins_pipeline.groovy]: Reverted this change.

**Comment by mseth10** [comment]
@mxnet-bot run ci [unix-cpu, unix-gpu]

**Comment by leezu** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by mseth10** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by mseth10** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by ciyongch** [comment]
Thanks @mseth10 for the backport. 
Ping @TaoLv , @pengzhao-intel , @leezu to have a review and help merge.

**Pull request merged**